x/evm/wrappers/utils.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package wrappers
+
+import (
+	"fmt"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/evmos/evmos/v20/x/evm/config"
+)
+
+// convertEvmCoinTo18DecimalsUnchecked convert the given coin from its original
+// representation into a 18 decimals one. The function panic if coin denom is
+// not the evm denom or in case of overflow.
+func mustConvertEvmCoinTo18Decimals(coin sdk.Coin) sdk.Coin {
+	if coin.Denom != config.GetEVMCoinDenom() {
+		panic(fmt.Sprintf("expected evm denom %s, received %s", config.GetEVMCoinDenom(), coin.Denom))
+	}
+
+	evmCoinDecimal := config.GetEVMCoinDecimals()
+	newAmount := coin.Amount.Mul(math.NewInt(evmCoinDecimal.ConversionFactor()))
+
+	return sdk.NewCoin(coin.Denom, newAmount)
+}
+
+// convertEvmCoinFrom18Decimals converts the coin's Amount from 18 decimals to its
+// original representation. Return an error if the coin denom is not the EVM
+// denom or in case of underflow.
+func convertEvmCoinFrom18Decimals(coin sdk.Coin) (sdk.Coin, error) {
```

Comment:
this function is only used in `convertCoinsFrom18Decimals` and we're checking twice the `if coin.Denom != config.GetEVMCoinDenom() `

Reply:
maybe we can remove this function and add the conversion directly on `convertCoinsFrom18Decimals`

Reply:
I would keep it for now, and refactor if we dont need it after adding the feemarket wrapper, and the 6 decimals tests